### ISSUE TITLE
Authentication required for pushing git tag

### ISSUE THREAD

**Issue by broth-eu** [vivin/gradle-semantic-build-versioning#73]
When I invoke task  `tag` with option `--push`, I always get the following error:
_Authentication is required but no CredentialsProvider has been registered_

How do I need to specify my credentials for the remote git repository?

**Comment by vivin** [comment]
I think this may be related to configuring git. Are you able to push tags manually from the command-line using git by itself?

**Comment by broth-eu** [comment]
Yes, pushing from command-line works as expected. Currently, my workaround is to push the newly created version tag by invoking the according git command.

**Comment by vivin** [comment]
I'll look into this a bit more. It might be related to the way JGit does things. I vaguely remember running into a similar issue before.

**Comment by vivin** [comment]
@broth-eu Are you trying to push to github? Or is it some other git repo? What sort of authentication are you using?

**Comment by broth-eu** [comment]
I'm trying to push to a BitBucket cloud repo. I always authenticate using username and password. Windows provides both using its credential manager.

**Comment by vivin** [comment]
I see; so every time you push, you provide your username and password?

**Comment by vivin** [comment]
I think a way around this is to be able to provide authentication information to the plugin and configure the credentials holder that way.

**Comment by vivin** [comment]
Sorry I've been really busy the past few months. Will try to look at this again.

**Comment by kazvictor** [comment]
I get the same error when this is run from TravisCI:
```
:tag FAILED
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':tag'.
> org.eclipse.jgit.api.errors.TransportException: https://github.com/example-org/my-repo.git: Authentication is required but no CredentialsProvider has been registered
```

How do you?
> provide authentication information to the plugin and configure the credentials holder

**Comment by kazvictor** [comment]
I fixed this in my use case for travisCI by pushing the tag separately:
```
script: ./gradlew printVersion tag -Prelease && git push "https://${GH_TOKEN}@github.com/${TRAVIS_REPO_SLUG}.git" --tags
```

**Comment by vivin** [comment]
I'm trying to fix another issue currently, but will look at this again. I'm not sure why it isn't pulling in auth settings.

**Comment by jsmithpkp** [comment]
I am having the same problem with gitlab and the git server and the CI server are the same. I had to do something similar to  kazvictor.
